server/services/warrantFormDataService.ts

```diff
@@ -0,0 +1,46 @@
+import { Request } from 'express'
+import WarrantFormData from '../model/WarrantFormData'
+import PrisonerService from './prisonerService'
+
+export default class WarrantFormDataService {
+  constructor(private readonly prisonerService: PrisonerService) {}
+
+  public setData(req: Request, nomsId: string, formData: WarrantFormData): void {
+    if (!req.session.formData) {
+      req.session.formData = {}
+    }
+    req.session.formData[nomsId] = formData
+  }
+
+  public resetData(req: Request, nomsId: string): void {
+    if (!req.session.formData) {
+      req.session.formData = {}
+    }
+    req.session.formData[nomsId] = undefined
+  }
+
+  public getData(req: Request, nomsId: string): WarrantFormData {
+    if (req.session.formData && req.session.formData[nomsId]) {
+      return req.session.formData[nomsId]
+    }
+    return {}
+  }
+
+  public async pushToNomis(req: Request, nomsId: string, userCaseLoads: string[], token: string): Promise<void> {
+    const data = this.getData(req, nomsId)
+    const prisoner = await this.prisonerService.getPrisonerDetail(nomsId, userCaseLoads, token)
+    const { bookingId } = prisoner
```

Comment:
Could this be inlined above? It doesn't look like `prisoner` is used again?

Reply:
Yea good point. I'll address it in the next one.